res/controllers/Pioneer-DDJ-400-script.js

```diff
@@ -0,0 +1,719 @@
+            data1: 0x47,
+        },
+        cue: {
+            status: 0x91,
+            data1: 0x0C,
+        },
+        shiftCue: {
+            status: 0x91,
+            data1: 0x48,
+        },
+    },
+};
+
+// Save the Shift State
+PioneerDDJ400.shiftState = [0, 0];
+
+// JogWheel
+PioneerDDJ400.vinylMode = true;
+PioneerDDJ400.alpha = 1.0 / 8;
+PioneerDDJ400.beta = PioneerDDJ400.alpha / 32;
+PioneerDDJ400.highspeedScale = 2;
+PioneerDDJ400.bendScale = 0.5;
+
+PioneerDDJ400.pointJumpSpace = 0.005; // amount in percent of the Song we can jump back to previous Cue or loop point
+
+PioneerDDJ400.tempoRanges = [0.06, 0.10, 0.16, 0.25]; // WIDE = 25%?
+
+// Keyboard Mode Variables and Settings
+PioneerDDJ400.keyboardHotCuePoint = [0, 0]; // selected HotCue point (eg. PAD) in Keyboard mode per Deck 0 = unset
+PioneerDDJ400.keyboardModeRefCount = [0, 0]; // count the currently pressed Pads per Deck
+PioneerDDJ400.halftoneToPadMap = [4, 5, 6, 7, 0, 1, 2, 3];
+
+// Beatjump Pads
+PioneerDDJ400.beatjumpPad = [-1, 1, -2, 2, -4, 4, -8, 8]; // < 0 = left; else right
+PioneerDDJ400.beatjumpMulitplier = [1, 1]; // Beatjump Beatcount per Deck
+
+// Hotcue Pads saved Loop points
+PioneerDDJ400.hotcueLoopPoints = {
+    '[Channel1]': [],
+    '[Channel2]': []
+};
+
+// Loop Section
+PioneerDDJ400.loopin4beat = [false, false]; // inn4loop is pressed
+PioneerDDJ400.loopout = [false, false]; // out loop is pressed
+PioneerDDJ400.loopAdjustMultiply = 5;
+
+
+PioneerDDJ400.init = function() {
+    // init controller
+    // init tempo Range to 10% (default = 6%)
+    engine.setValue('[Channel1]', 'rateRange', PioneerDDJ400.tempoRanges[1]);
+    engine.setValue('[Channel2]', 'rateRange', PioneerDDJ400.tempoRanges[1]);
+
+    // enable effect focus on FX3 for Effect Section
+    engine.setValue('[EffectRack1_EffectUnit1]', 'show_focus', 0);
+    engine.setValue('[EffectRack1_EffectUnit1]', 'group_[Channel1]_enable', 1);
+
+    engine.setValue('[EffectRack1_EffectUnit2]', 'show_focus', 0);
+    engine.setValue('[EffectRack1_EffectUnit2]', 'group_[Channel2]_enable', 1);
+
+    engine.setValue('[EffectRack1_EffectUnit3]', 'show_focus', 1);
+
+    // Connect the VU-Meter LEDS
+    engine.connectControl('[Channel1]', 'VuMeter', 'PioneerDDJ400.vuMeterUpdate');
+    engine.connectControl('[Channel2]', 'VuMeter', 'PioneerDDJ400.vuMeterUpdate');
+
+    // reset vumeter
+    PioneerDDJ400.toggleLight(LightsPioneerDDJ400.deck1.vuMeter, false);
+    PioneerDDJ400.toggleLight(LightsPioneerDDJ400.deck2.vuMeter, false);
+};
+
+PioneerDDJ400.toggleLight = function(midiIn, active) {
+    midi.sendShortMsg(midiIn.status, midiIn.data1, active ? 0x7F : 0);
+};
+
+PioneerDDJ400.jogTurn = function(channel, _control, value, _status, group) {
+    var deckNum = channel + 1;
+    // wheel center at 64; <64 rew >64 fwd
+    var newVal = value - 64;
+
+    // loop_in / out adjust
+    var loopEnabled = engine.getValue(group, 'loop_enabled');
+    if (loopEnabled > 0) {
```

Comment:
Remove this. `loop_in` and `loop_out` take care of this if you use them correctly by setting them to 1 on button press and 0 on button release.